### PR TITLE
docs(contributing): document version range policy for dependencies

### PR DESCRIPTION
**Pull request by basti1302**
Also, add docs on how to update a single version in a package and
rearrange the dependency version management sections.

refs 120539

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ The following sections describe how to manage dependencies in practice.
 
 ### package-lock.json
 
+Note: The `package-lock.json` files are only relevant when you execute an `npm install` in _this repository_. Users of `@instana` npm packages will have their own `package-lock.json` or `yarn.lock` file for their application, those are completely independent from the `package-lock.json` files in this repository. Thus, they need to take care of updating their transitive dependencies themselves. We only can make sure that the version ranges we use for our direct dependencies (in our `package.json` files) allow to install a dependency tree that has no known vulnerabilities.
```
Comment (basti1302) [inline CONTRIBUTING.md:74]: This was previously in the section "Updating A Single Version In A Lockfile", but it makes more sense here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -90,18 +92,20 @@ Do *not* run `npm install ${dependency-name}` in the directory of a package (lik
...
 
-### Removing A Package Dependency
-
-Do *not* run `npm uninstall ${dependency-name}` in the directory of a package. This will mess up the `package-lock.json` file in that package.
-
-There is no `lerna` command to remove a dependency, that is, there is no `lerna remove --scope` counterpart for `lerna add --scope`. To remove a dependency from a package, remove the corresponding entry from the `dependencies`/`devDependencies`/... section of the package's `package.json` file and then run `npm run refresh-package-lock-files` afterwards. This will remove the package from the `node_modules` folder and also update the `package-lock.json` file correctly.
-
```
Comment (basti1302) [inline CONTRIBUTING.md:98]: Moved this section down under the sections on updating dependencies, without changing the content.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -117,7 +121,31 @@ npm run refresh-package-lock-files
 # Finally, check that the dependency in question has been updated in `packages/shared-metrics/package-lock.json` to the desired version.
 ```
 
-Note: The intent of the procedure above is satisfy security scanners (`npm audit`, Dependabot etc.). `package-lock.json` files are never included when publishing to the npm registry. Thy are only relevant when you execute an `npm install` in _this repository_. Users of `@instana` npm packages will have their own `package-lock.json` or `yarn.lock` file for their application, those are completely independent from the `package-lock.json` files in this repository. Thus, they need to take care of updating their transitive dependencies themselves. We only can make sure that the version ranges we use for our direct dependencies allow to install a dependency tree that has no known vulnerabilities.
```
Comment (basti1302) [inline CONTRIBUTING.md:120]: Most of this has been moved up, so we now only link to the section where this is explained.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -90,18 +92,20 @@ Do *not* run `npm install ${dependency-name}` in the directory of a package (lik
...
 You can run `npm run update-deps` to update all dependencies in all packages via [`npm-check-updates`](https://www.npmjs.com/package/npm-check-updates) in one batch.
 
+#### Updating A Single Version In A `package.json` File
+
+Run `lerna add --scope=@instana/$package $dependencyName@$version && npm run refresh-package-lock-files`. For example, when you want to update the `semver` dependency in `@instana/core` to version 7.5.3, run `lerna add --scope=@instana/core semver@7.5.3 && npm run refresh-package-lock-files`.
+
+You can manage dev dependencies in the same way, that is `lerna add --dev --scope=@instana/autoprofile semver@7.5.3 && npm run refresh-package-lock-files`.
+
+Do *not* run `npm install ${dependency-name}` in the directory of a package (like, in `packages/core`) to update a dependency. This will mess up the `package-lock.json` file in that package. (If you accidentally did this, you can do a `git checkout` of the `package.json` file of the package and then run `npm run refresh-package-lock-files` to fix the lock file and the content of `node_modules`.)
+
```
Comment (basti1302) [inline CONTRIBUTING.md:108]: This is new content.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -117,7 +121,31 @@ npm run refresh-package-lock-files
...
+Note: The intent of the procedure above is satisfy security scanners (`npm audit`, Dependabot etc.) we run ourselves. The `package-lock.json` files are never included when publishing to the npm registry, see section <#package-lock.json>.
+
+### Removing A Package Dependency
+
+Do *not* run `npm uninstall ${dependency-name}` in the directory of a package. This will mess up the `package-lock.json` file in that package.
+
+There is no `lerna` command to remove a dependency, that is, there is no `lerna remove --scope` counterpart for `lerna add --scope`. To remove a dependency from a package, remove the corresponding entry from the `dependencies`/`devDependencies`/... section of the package's `package.json` file and then run `npm run refresh-package-lock-files` afterwards. This will remove the package from the `node_modules` folder and also update the `package-lock.json` file correctly.
+
+#### Version Ranges vs. Pinning a Specific Version
```
Comment (basti1302) [inline CONTRIBUTING.md:132]: This section is also new. It is one half of https://instana.kanbanize.com/ctrl_board/56/cards/120539/details/ (without actually changing all dependencies from pinnned versions ranges, that needs to be done later)

**Review by kirrg001 — APPROVED** [review]

**Pull request merged**